src/middleware/router.js

```diff
@@ -488,6 +493,11 @@ function sendRequest(ctx, route, options) {
       }
     }
 
+    if (route.primary) {
+      // Save request body of the primary route
+      ctx.reqBody = Buffer.isBuffer(ctx.body) ? ctx.body.toString() : ctx.body
```

Comment:
Wouldn't this only be set if the Kafka route is a secondary route. What happen if the Kafka route is a primary route?